h2o-core/src/main/java/water/H2O.java

```diff
@@ -843,7 +848,17 @@ else if(s.matches(("client_disconnect_timeout"))){
         i = s.incrementAndCheck(i, args);
         String value = args[i];
         trgt.extra_headers = ArrayUtils.append(trgt.extra_headers, new KeyValueArg(key, value));
-      } else if(s.matches("embedded")) {
+      } else if (s.matches("file_deny_glob")) {
+        i = s.incrementAndCheck(i, args);
+        String key = args[i];
+        try {
+          trgt.file_deny_glob = FileSystems.getDefault().getPathMatcher("glob:" + key);
+        }
+        catch (Exception e) {
+          throw new IllegalArgumentException("Error parsing file_deny_glob parameter");
+        }
+      }
+      else if(s.matches("embedded")) {
```

Comment:
@krasinski :  This seems to significantly slow down H2O.  Is it possible to do this check less frequently?  Like when there is a actual reading/writing that is about to happen rather than any time someone wants to look at the key?

Reply:
@krasinski :  It seems like my previous comment was wrong.  The failure is probably caused by error throws by worker node not caught and the whole test just hang.  Here is a copy of the test run time:

![image](https://github.com/h2oai/h2o-3/assets/7231712/f1e64b2c-a439-4417-9e99-77de78cf1f6b)